### DISCUSSION TITLE
각종 패키지관리자 등재 여부

### DISCUSSION THREAD

**Discussion by gintering1000** [orioncactus/pretendard#49] [General]
관련 패키지 관리자
- 윈도우용 : chocolatey
- 맥용 : brew
- 우분투용 :  apt-get

[이전 이슈글](https://github.com/orioncactus/pretendard/issues/46)

**Comment by gintering1000** [comment]
안녕하세요!

# 패키지 관리자 등재는
보통 소스코드 내부에 등재설정파일을 넣어두고 해당 패키지관리자에 등록하는 방법이 사용됩니다.
하지만 가장 일반적인 방법은 brew에서 폰트 관리를 하는 cask-font 라는 패키지가 이미 있고 해당 패키지 주인에게 pretended를 넣어달라고 하는게 가장 좋은 방법 같습니다.

# 하지만 생각해보니!
나눔폰트의 등재는 구글폰트사이트의 주소를 이용해 등재가 되었고~
pretended는 github주소를 기준으로 할것인지~
구글폰트에 등록되면 등록 후의 주소로 할것인지 ( 이것이 더 장기적으로 나을 듯 합니다 )
를 고민해보시고 전략적으로 선택해보시면 될듯합니다.


# brew-nanum-gothic의 패키지 정보
![image](https://user-images.githubusercontent.com/77244047/148693824-74ae654d-301f-4038-b28b-b111dd7ccdf9.png)

# 사람들이 보통 brew를 통해 설치하는 방법
![image](https://user-images.githubusercontent.com/77244047/148693831-2a941a71-d56f-45e9-9931-3ae19d90cfa0.png)

# 기타
homebrew-cask-fonts 주소 : https://github.com/Homebrew/homebrew-cask-fonts
homebrew-cask-fonts / 나눔예시 : https://github.com/Homebrew/homebrew-cask-fonts/blob/HEAD/Casks/font-nanum-gothic.rb

**Comment by kms0219kms** [comment]
homebrew 등재는 구글 폰트에 등재되면 request가 가능합니다.
따라서, 구글 폰트 등재되도록 제작 중이시니 구글 폰트에 등재되면 진행하면 될 것 같습니다. (#7 #46 참조)

**Comment by orioncactus** [comment]
안녕하세요! 패키지에 관해 자세한 지식이 없어 확인하느라 답변이 늦었습니다. 기다려주셔서 감사드리며, 이렇게 자세히 설명해주시고 신경써주셔서 진심으로 감사드립니다.

@kms0219kms님께서 말씀해주신 것처럼, homebrew-cask-fonts에 등록된 글꼴들을 살펴보니 Google Fonts에 등록된 글꼴로부터 제공하는 방법으로 관리가 되고 있는 듯합니다. 추후 Pretendard에 CJK 적용을 마치고 안정화되었을 때 Google Fonts에 등록을 고려하고 있는데, 시간이 조금 걸려도 괜찮으시다면 그때까지 기다려주시거나, 그보다 더 일찍 사용하고자 하신다면 cask-fonts와는 별개로 brew에 패키지를 등록해 올리는 방법이 있을 듯합니다.

제가 보기에는 Google Fonts에 등록할 때 이어 진행하면 어떨까 싶습니다. 빠르게 사용하기 위한 방법을 고려하시면 저도 확인해보겠습니다. 감사합니다!

**Reply by gintering1000** [reply]
아닙니다. 패키지가 분산되면 미래에 더욱 복잡해질듯합니다. 통일성을 위해서 기다리겠습니다

**Reply by orioncactus** [reply]
@sudosubin 님의 기여 덕분에 이제 homebrew-cask-fonts 및 nix에서 Pretendard를 사용하실 수 있습니다 :fire: 어차피 Pretendard는 계속해서 업데이트될 예정이기 때문에, Google Fonts에 등록하기 전에 미리 homebrew-cask-fonts에서 등록되는 것도 괜찮아 보입니다. 자세한 내용은 아래에서 확인하실 수 있습니다 :D
- https://github.com/orioncactus/pretendard#시스템-폰트

**Comment by kms0219kms** [comment]


**Reply by gintering1000** [reply]
whatever you want, master~
